cpp/include/cuspatial_test/geometry_generator.cuh

```diff
@@ -0,0 +1,255 @@
+/*
+ * Copyright (c) 2023, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <cuspatial_test/vector_factories.cuh>
+
+#include <cuspatial/cuda_utils.hpp>
+#include <cuspatial/error.hpp>
+#include <cuspatial/experimental/ranges/multipolygon_range.cuh>
+#include <cuspatial/vec_2d.hpp>
+
+#include <rmm/cuda_stream_view.hpp>
+#include <rmm/device_uvector.hpp>
+#include <rmm/exec_policy.hpp>
+
+#include <thrust/sequence.h>
+#include <thrust/tabulate.h>
+
+namespace cuspatial {
+namespace test {
+
+constexpr double PI = 3.14159265358979323846;
+
+/**
+ * @brief Struct to store the parameters of the multipolygon array generator
+ *
+ * @tparam T Type of the coordinates
+ */
+template <typename T>
+struct multipolygon_generator_parameter {
+  using element_t = T;
+
+  std::size_t num_multipolygons;
+  std::size_t num_polygons_per_multipolygon;
+  std::size_t num_holes_per_polygon;
+  std::size_t num_sides_per_ring;
+  vec_2d<T> centroid;
+  T radius;
+
+  CUSPATIAL_HOST_DEVICE std::size_t num_polygons()
+  {
+    return num_multipolygons * num_polygons_per_multipolygon;
+  }
+  CUSPATIAL_HOST_DEVICE std::size_t num_rings() { return num_polygons() * num_rings_per_polygon(); }
+  CUSPATIAL_HOST_DEVICE std::size_t num_coords() { return num_rings() * num_vertices_per_ring(); }
+  CUSPATIAL_HOST_DEVICE std::size_t num_vertices_per_ring() { return num_sides_per_ring + 1; }
+  CUSPATIAL_HOST_DEVICE std::size_t num_rings_per_polygon() { return num_holes_per_polygon + 1; }
+  CUSPATIAL_HOST_DEVICE T hole_radius() { return radius / (num_holes_per_polygon + 1); }
+};
+
+/**
+ * @brief Generate coordinates for the ring based on the local index of the point.
+ *
+ * The ring is generated by walking a point around a centroid with a fixed radius.
+ * Each step has equal angles.
+ *
+ * @tparam T Type of coordinate
+ * @param point_local_idx Local point of the point
```

Comment:
Please clarify. What does "Local point of the point" mean?

Reply:
`local index of the point`